stdlib/src/collections/list.mojo

```diff
@@ -197,6 +197,60 @@ struct List[T: CollectionElement](
     # Operator dunders
     # ===-------------------------------------------------------------------===#
 
+    fn __eq__[
```

Comment:
Question:
I'm not very familiar with conditional conformance yet
```mojo
fn __eq__[T2: ComparableCollectionElement](self: List[T], other: List[T2])
```
can't they just be set to be the same type? or does it not work?
```mojo
fn __eq__[C: ComparableCollectionElement](self: List[C], other: List[C])
```

Reply:
I think `fn __eq__[T: ComparableCollectionElement](self: List[T], other: List[T]) -> Bool` should work, and we don't need to use the rebind trick.

Reply:
If we define `__eq__` as follows
```mojo
fn __eq__[T: C...](self: List[T], other: List[T]) -> Bool:
```
then `a.__eq__(b)` would work, while `a == b` would throw the following error
```
invalid call to `__eq__`: could not deduce parameter `T` of parent struct `List`
```

Reply:
@lattner is this something we can fix parameter inference wise so we can retain the nice sugar syntax of `==` without dropping back to `__eq__` manually?  I also ran into this yesterday when looking at removing some rebinds we were talking about.

Reply:
FYI, I raised #3199 for this.

Reply:
Using `==` instead of `__eq__` should work fine now since https://github.com/modularml/mojo/issues/3199 is closed now - see the discussion there.